velox/connectors/hive/HiveDataSink.cpp

```diff
@@ -394,7 +438,11 @@ void HiveDataSink::appendData(RowVectorPtr input) {
     RowVectorPtr writerInput = partitionSize == input->size()
         ? input
         : exec::wrap(partitionSize, partitionRows_[index], input);
-    write(index, writerInput);
+    auto dataInput = filterPartitionColumns(
+        getPartitionChannels(insertTableHandle_),
+        writerInput,
+        connectorQueryCtx_->memoryPool());
+    write(index, dataInput);
```

Comment:
It looks like a repeated pattern that you call filterPartitionColumns() before calling write(index, dataInput). Does it make sense if we put filterPartitionColumns() into the body of write()? It looks to me your filterPartitionColumns() also works well when getPartitionChannels() is empty vector.